app/controllers/previousRegistrations/AddPreviousRegistrationController.scala

```diff
@@ -88,15 +89,28 @@ class AddPreviousRegistrationController @Inject()(
               formWithErrors =>
                 Future.successful(BadRequest(view(formWithErrors, waypoints, previousRegistrations, canAddCountries))),
 
-              value =>
+              (addAnotherRegistration: Boolean) =>
                 for {
-                  updatedAnswers <- Future.fromTry(request.userAnswers.set(AddPreviousRegistrationPage(), value))
+                  updatedAnswers <- Future.fromTry(request.userAnswers.set(AddPreviousRegistrationPage(), addAnotherRegistration))
                   _ <- cc.sessionRepository.set(updatedAnswers)
-                } yield Redirect(AddPreviousRegistrationPage().navigate(waypoints, request.userAnswers, updatedAnswers).url)
+                } yield Redirect(AddPreviousRegistrationPage().navigate(
+                  calculateNextStepWaypoints(waypoints, addAnotherRegistration),
+                  request.userAnswers,
+                  updatedAnswers).url
+                )
             )
         }
       }
   }
 
+  private def calculateNextStepWaypoints(waypoints: Waypoints, addAnotherRegistration: Boolean) = {
```

Comment:
Fix navigation buck in check/amend mode

Reply:
Why do we need to do this here, do we not know the state during the page navigation? I guess would be similar to this logic https://github.com/hmrc/ioss-registration-frontend/pull/44/files#diff-5e7a8272d4e8f574e76a49f13151151f82d0260c93d2cee12cfa2fe5bdc10e0bR43